File: app/src/main/java/com/github/overpass/gather/auth/login/forgot/PasswordResetRepo.java
package com.github.overpass.gather.auth.login.forgot;

import com.google.firebase.auth.FirebaseAuth;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

public class PasswordResetRepo {

    private final FirebaseAuth firebaseAuth;

    public PasswordResetRepo(FirebaseAuth firebaseAuth) {
        this.firebaseAuth = firebaseAuth;
    }

    public LiveData<ForgotStatus> sendForgotPassword(String email) {
        MutableLiveData<ForgotStatus> result = new MutableLiveData<>();
        result.setValue(new ForgotStatus.Progress());
        firebaseAuth.sendPasswordResetEmail(email)
                .addOnSuccessListener(__ -> {
                    result.setValue(new ForgotStatus.Success());
                })
                .addOnFailureListener(e -> {
                    result.setValue(new ForgotStatus.Error(e));
                });
        return result;
    }
}

File: app/src/main/java/by/overpass/gather/model/repo/password/PasswordResetRepo.kt
package by.overpass.gather.model.repo.password

import com.google.android.gms.tasks.Task
import com.google.firebase.auth.FirebaseAuth
import javax.inject.Inject

class PasswordResetRepo @Inject constructor(private val firebaseAuth: FirebaseAuth) {

    fun sendForgotPassword(email: String): Task<Void> = firebaseAuth.sendPasswordResetEmail(email)
}
